### PR TITLE
Improve support for sinope Calypso and Eco Mode

### PR DESCRIPTION
**Pull request by claudegel**
## Proposed change

- Code cleanup for calypso controller.
- Add support for Eco Mode on thermostats, Eco Sinopé
- Add  many attributes for sensor and light in manufacturer cluster

## Checklist

- [x] The changes are tested and work correctly
- [x] `pre-commit` checks pass / the code has been formatted using Black
- [ ] Tests have been added to verify that the new code works

### CODE REVIEW

File: zhaquirks/sinope/switch.py [hunk 1]
```diff
@@ -127,12 +119,14 @@ class FlowDuration(t.uint32_t):
         0x0003: ("firmware_number", t.uint16_t, True),
         0x0004: ("firmware_version", t.CharacterString, True),
         0x0010: ("outdoor_temp", t.int16s, True),
-        0x0013: ("tank_size", TankSize, True),
+        0x0013: ("unknown_attr_1", t.enum8, True),
```
Comment (TheJulianJES) [inline zhaquirks/sinope/switch.py:122]: Just to confirm, this attribute wasn't actually the `TankSize`, right?

**Comment by claudegel** [comment]
It was an error as the TankSize finally is not in the attribute but only in Neviweb cloud. It is not used by the Calypso. I've left the attribute there as other could be able to find out the usage.
I have many other attribute that are still unknown. is it better to leave them out until I find the usage or put them in the quirks so other have access to play with them.

**Comment by claudegel** [comment]
Another thing, I'm thinking of splitting the 4 files into specific file for each device model. Could it be better ? It's a big job to separate everything but if it is better I can try to split them.

**Comment by TheJulianJES** [comment]
> I'm thinking of splitting the 4 files into specific file for each device model.

Generally yeah, it's nicer to have different device types separated. For example, `switch.py` has smart plugs, valve controllers, ... from what I can see. IMO, it's still *fine* for these, as there aren't too many models, but we're open for a "cleanup PR" if you want.

**Review by TheJulianJES — APPROVED** [review]
LGTM

**Comment by claudegel** [comment]
I have many attributes found in manufacturer cluster but still need to find out the use. Generally I don't add them to the quirks but could it be better to add them as unknown_attr_xx so other can try to understand the usage for them so we can put a name on them ?

**Comment by TheJulianJES** [comment]
> many attributes found in manufacturer cluster but still need to find out the use. Generally I don't add them to the quirks but could it be better to add them as unknown_attr_xx

I don't have a real preference. If they're likely not needed, I wouldn't add them, but it should also be fine if you do add them.

**Pull request merged**